Developer-DAO/DAO-job-board
# refactor: implements new JobCard design

## Description
This PR takes the new Job Card design from the [Job Board Figma File](https://www.figma.com/file/qGlR5rEinlbUgWmbogXunz/Design-System-%2F-Jobs-Board?node-id=444%3A2525).

### Linked Issue: #155 

#### Notes
- The favorite icon does not include the gradient from the design due to not wanting to spend more time than allotted for this feature. I hope to go back at a later date, or we create another issue for this specifically, and figure out how to go about this with the `react-fontawesome` package.
- At the moment, the Organization data is hard coded due to my lack of knowledge on how the queries for both the job and the related organization data will be passed in.

## Validation Steps
1. navigate to `/jobs`
2. Verify that you see the new `<JobCard />` design
![image](https://user-images.githubusercontent.com/118162/145925308-98b73088-f402-4899-8522-c83dbbb10bf8.png)
3. Click on the heart icon to "un-favorite" the job
![image](https://user-images.githubusercontent.com/118162/145925506-34f1dfdc-a104-476a-9270-2b11482419ba.png)
4. Verify that the icon is no longer filled and there is a color change
![image](https://user-images.githubusercontent.com/118162/145925522-779eaee5-cc21-4c3e-8806-3a865f2fc100.png)
